src/language/class-modifiers.md

```diff
@@ -0,0 +1,258 @@
+implementation. Abstract classes often have [abstract methods][].
+
+```dart
+// Library a.dart
+abstract class Vehicle { 
+  void moveForward();     // Abstract method.
+}
+```
+
+```dart
+// Library b.dart
+import 'a.dart';
+
+var myCar = Vehicle();     // Error: Cannot be constructed
+
+class Car extends          // Can be extended
+  Vehicle {
+    int passengers;
+    ...
+}
+
+class MockVehicle          // Can be implemented
+  implements Vehicle {
+    @override
+    void moveForward(int meters) { ... }
+}
+```
+
+If you want your abstract class to appear to be instantiable,
+define a [factory constructor][].
+
+## `base`
+
+Use the `base` modifier to enforce inheritance of a class or mixin's implementation.
+The `base` modifier disallows implementation outside of the current library, which
+guarantees:
+
+- The base class constructor is called whenever an instance of a subtype of the
+class is created.
+- That all implemented private members exist in subtypes.
+- Adding a new implemented member to a `base` class will not break subtypes (except
+  when they already declare a member with the same name and an incompatible
+  signature), since all subtypes will inherit the new member.
+
+The `base` modifier must be applied transitively to all the subtypes of a class;
+any class which implements or extends a base class must be marked `base`, `final`,
+or `sealed`. 
+
+```dart
+// Library a.dart
+base class Vehicle {
+  void moveForward(int meters) { ... }
+}
+```
+
+```dart
+// Library b.dart
+import 'a.dart';
+
+var myCar = Vehicle();          // Can be constructed
+
+base class Car extends          // Can be extended
+  Vehicle {
+    int passengers;
+    ...
+}
+
+base class MockVehicle          // ERROR: Cannot be implemented
+  implements Vehicle {
+    @override
+    void moveForward ...
+}
+```
+
+## `interface`
+
+Use the `interface` modifier to define an interface. Interface classes can be
+implemented outside of the current library, but not extended. This guarantees:
+
+- All members of the class must have an implementation.
+- Users of the API are prevented from overriding a subset of methods.
+- A solution to the [fragile base class problem][], or the freedom to change the
+interface's implementation, because there are no inherited member implementations
+outside of the current library whose behavior needs to be preserved. 
+
+```dart
+// Library a.dart
+interface class Vehicle {
+  void moveForward(int meters) { ... }
+}
+```
+
+```dart
+// Library b.dart
+import 'a.dart';
+
+var myCar = Vehicle();     // Can be constructed
+
+class Car extends          // ERROR: Cannot be inherited
+  Vehicle {
+    int passengers;
+    ...
+}
+
+class MockVehicle          // Can be implemented     
+  implements Vehicle {
+    @override
+    void moveForward ...
+}
+```
+
+To define a pure interface that others are free to implement, that cannot be
+inherited but is allowed to have abstract members, define an
+`abstract interface class`.
```

Comment:
i feel like we should emphasize this combination more. maybe even having this as a top-level section before the existing `interface class`?